plugins/webp-uploads/tests/test-picture-element.php

```diff
@@ -436,10 +436,15 @@ public function test_disable_responsive_image_with_picture_element( Closure $set
 		$this->assertStringEndsWith( '.jpg', $picture_processor->get_attribute( 'src' ), 'Make sure the fallback IMG should return JPEG src.' );
 
 		$picture_processor = new WP_HTML_Tag_Processor( $picture_markup );
+		$source_count      = 0;
 		while ( $picture_processor->next_tag( array( 'tag_name' => 'source' ) ) ) {
+			++$source_count;
 			$this->assertSame( self::$mime_type, $picture_processor->get_attribute( 'type' ), 'Make sure the Picture source should not return JPEG as source.' );
 			$this->assertStringContainsString( '.webp', $picture_processor->get_attribute( 'srcset' ), 'Make sure the Picture source srcset should return WEBP images.' );
 		}
+
+		// Ensure at least one source element was created (addresses GitHub issue with wp_calculate_image_srcset disabled).
```

Comment:
it would be nice to add a separate test that sets the wp_calculate_image_srcset filter to return false and verifies the source generation.